src/frequenz/sdk/microgrid/_data_pipeline.py

```diff
@@ -433,6 +471,28 @@ def logical_meter() -> LogicalMeter:
     return _get().logical_meter()
 
 
+def consumer() -> Consumer:
+    """Return the consumer instance.
+
+    If a Consumer instance doesn't exist, a new one is created and returned.
+
+    Returns:
+        A Consumer instance.
```

Comment:
Same here, if we make it a property then it is also easier to get rid of the redundancy.